actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -151,6 +154,14 @@ def valid_accept_header
       def use_accept_header
         !self.class.ignore_accept_header
       end
+
+      private
+        def extract_variant!
+          if parameters.key?(:format)
+            parameters[:format], *variants = parameters[:format].to_s.split("+")
+            self.variant = variants.map(&:to_sym) unless variants.empty?
```

Comment:
Not sure what you mean, `unless variants.empty?` and `if variants.any?` are logically identical